backend/src/apiserver/server/run_server.go

```diff
@@ -40,6 +40,48 @@ func (s *RunServer) CreateRun(ctx context.Context, request *api.CreateRunRequest
 	return ToApiRunDetail(run), nil
 }
 
+func (s *RunServer) ResubmitRun(ctx context.Context, request *api.ResubmitRunRequest) (*api.RunDetail, error) {
+	// Get old run details.
+	runDetails, err := s.resourceManager.GetRun(request.Id)
+	if err != nil {
+		return nil, err
+	}
+
+	var workflow util.Workflow
+	if err := json.Unmarshal([]byte( runDetails.WorkflowRuntimeManifest), &workflow); err != nil {
+		return nil, util.NewInternalServerError(err, "Failed to retrieve the runtime pipeline spec from the old run")
+	}
+
+	// Formulate the new argo workflow.
+	newWorkflow, err := formulateResubmitWorkflow(workflow.Workflow, s.resourceManager.GetRandomString())
+	if err != nil {
+		return nil, util.NewInternalServerError(err, "Failed to resubmit the old run.")
+	}
+
+	workflowManifest, err := json.Marshal(newWorkflow)
+	if err != nil {
+		return nil, util.NewInternalServerError(err, "Failed to resubmit the old run.")
+	}
+
+	parameters, err := toApiParameters(runDetails.PipelineSpec.Parameters)
+	if err != nil {
+		return nil, util.NewInternalServerError(err, "Failed to resubmit the old run.")
+	}
+
+	newRun := &api.Run{
+		Name:               request.Name,
+		Description:        runDetails.Description,
+		ResourceReferences: ToApiResourceReferences(runDetails.ResourceReferences),
+		PipelineSpec:       &api.PipelineSpec{WorkflowManifest: string(workflowManifest), Parameters: parameters},
+	}
+
+	run, err := s.resourceManager.CreateRun(newRun)
```

Comment:
As discussed, update existing run instead of creating a new run.

Reply:
done. please also see PR description for changes on Persistence agent.